src/rust/plugin-work-queue/src/test_utils.rs

```diff
@@ -64,6 +69,31 @@ impl PsqlQueueTestExtensions for PsqlQueue {
         .await?;
         Ok(generator_messages)
     }
+
+    async fn get_all_analyzer_messages(
+        &self,
+        plugin_id: Uuid,
+    ) -> Result<Vec<NextExecutionRequest>, PsqlQueueError> {
+        let analyzer_messages = sqlx::query_as!(
+            NextExecutionRequest,
+            r#"
+            SELECT
+                 execution_key AS "execution_key!: ExecutionId",
+                 plugin_id,
+                 pipeline_message,
+                 tenant_id,
+                 trace_id,
+                 event_source_id
+            FROM plugin_work_queue.analyzer_plugin_executions
+            WHERE plugin_id = $1
+            "#,
+            plugin_id
+        )
+        .fetch_all(&self.pool)
+        .await?;
+
+        Ok(analyzer_messages)
+    }
 }
 
 pub async fn scan_for_plugin_message_in_pwq(
```

Comment:
another "oh fuck, i didn't name something forward-thinking-ly": could we rename this to `scan_generator_messages`?